cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java

```diff
@@ -396,13 +416,30 @@ protected void addModule(CommandContext ctx, final ParsedCommandLine parsedCmd)
             }
         }
 
-        if(config != null) {
+        if (config != null) {
+            Set<String> modules = new HashSet<>();
             final String dependenciesStr = dependencies.getValue(parsedCmd);
             if(dependenciesStr != null) {
                 final String[] depsArr = dependenciesStr.split(",+");
                 for(String dep : depsArr) {
                     // TODO validate dependencies
                     config.addDependency(new ModuleDependency(dep));
+                    modules.add(dep);
```

Comment:
I didn't wanted to change the existing behavior. We could have customer scripts that wrongly add dependencies twice.